keras_nlp/models/deberta_v3/deberta_v3_tokenizer.py

```diff
@@ -120,6 +122,11 @@ def token_to_id(self, token):
             return self.mask_token_id
         return super().token_to_id(token)
 
+    def detokenize(self, ids):
+        blank_token_id = self.token_to_id("")
+        ids = tf.where(ids == self.mask_token_id, blank_token_id, ids)
```

Comment:
We make frequent use of `tf.ragged.boolean_mask` to do stuff like this.

Would `ids = tf.ragged.boolean_mask(ids, tf.not_equal(ids, self.mask_token_id))` work?

Also please add a unit test to verify this behavior.

Reply:
I think you just don't need the blank ids at all anymore. There is no need for a default value, essentially what you are doing is passing a mask of all location that should be kept to the boolean mask function.

Did you try the `ids = tf.ragged.boolean_mask(ids, tf.not_equal(ids, self.mask_token_id))` line? Remove `blank_token_id` entirely.